src/main/java/com/puppycrawl/tools/checkstyle/site/ClassAndPropertiesSettersJavadocScraper.java

```diff
@@ -0,0 +1,168 @@
+
+package com.puppycrawl.tools.checkstyle.site;
+
+import java.beans.Introspector;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.puppycrawl.tools.checkstyle.FileStatefulCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck;
+import com.puppycrawl.tools.checkstyle.utils.BlockCommentPosition;
+import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
+
+/**
+ * Class for scraping class javadoc and all property setter javadocs from the
+ * given checkstyle module.
+ */
+@FileStatefulCheck
+public class ClassAndPropertiesSettersJavadocScraper extends AbstractJavadocCheck {
+
+    /**
+     * Map of scraped javadocs - name of property, javadoc detail node.
+     * The class javadoc is stored too, with the key being the module name.
+     */
+    private static final Map<String, DetailNode> JAVADOC_FOR_MODULE_OR_PROPERTY =
+            new LinkedHashMap<>();
+
+    /** List of setter methods that the scraper ignores but shouldn't. */
+    private static final Set<String> SUPERCLASS_SETTERS = Set.of(
+        "AbstractFileSetCheck.setFileExtensions",
+        "AbstractHeaderCheck.setHeader"
+    );
+
+    /** Name of the module being scraped. */
+    private static String moduleName = "";
+
+    /**
+     * Initialize the scraper. Clears static context and sets the module name.
+     *
+     * @param newModuleName the module name.
+     */
+    public static void initialize(String newModuleName) {
+        clear();
+        moduleName = newModuleName;
+    }
+
+    /** Clear all scraped information. */
+    private static void clear() {
+        JAVADOC_FOR_MODULE_OR_PROPERTY.clear();
+        moduleName = null;
+    }
+
+    /**
+     * Get the module or property javadocs map.
+     *
+     * @return the javadocs.
+     */
+    public static Map<String, DetailNode> getJavadocsForModuleOrProperty() {
+        return Collections.unmodifiableMap(JAVADOC_FOR_MODULE_OR_PROPERTY);
+    }
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC,
+        };
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode ast) {
+        final DetailAST blockCommentAst = getBlockCommentAst();
+        if (BlockCommentPosition.isOnMethod(blockCommentAst)) {
+            final DetailAST methodDef = getParentAst(blockCommentAst, TokenTypes.METHOD_DEF);
+            if (methodDef != null) {
+                final String methodName = methodDef.findFirstToken(TokenTypes.IDENT).getText();
+                if (CheckUtil.isSetterMethod(methodDef) && isMethodOfScrapedModule(methodDef)
```

Comment:
@stoyanK7 @romani we have a conflict with https://github.com/checkstyle/checkstyle/pull/13607

Reply:
Not sure what to do here. Can `isSetterMethod ` be moved to `TokenUtil`? It has a few methods such as  `isBooleanLiteralType`, `isTypeDeclaration`, `isCommentType`..

Reply:
Please copy isSetterMethod to scrapper.
Let's not deal with dependencies for now.
Setter is very specific term that point of different interpretations.

Reply:
I would be fine to leave it in checkutil, let's see what @romani thinks